notebooks/demo_search.ipynb

```diff
@@ -157,6 +157,7 @@
    "outputs": [],
    "source": [
     "test_query = \"lighthouse\"\n",
+    "# test_query = \"wonder that features plants\"\n",
     "results = bm25_search_init.bm25_search(test_query, top_k=3)\n",
```

Comment:
Interesting set of results with this one, do you get the same?


Calculating embeddings: 100%|██████████| 1/1 [00:00<00:00, 58.87it/s]
**<ins>Hanging Gardens of Babylon - Score: 0.9838709677419354</ins>**
to have been built in the ancient city of Babylon, near present-day Hillah, Babil province, in Iraq. The Hanging Gardens' name is derived from the Greek word κρεμαστός (kremastós, lit. 'overhanging'), which has a broader meaning than the modern English word 'hanging' and refers to trees being planted on a raised structure such as a terrace.
Great Pyramid of Giza - Score: 0.5
The Great Pyramid of Giza is the largest Egyptian pyramid. It served as the tomb of pharaoh Khufu, who ruled during the Fourth Dynasty of the Old Kingdom. Built c. 2600 BC, over a period of about 26 years, the pyramid is the oldest of the Seven Wonders of the Ancient World, and the only wonder that has remained largely intact. It is the most 
**<ins>Hanging Gardens of Babylon - Score: 0.5</ins>**
The Hanging Gardens of Babylon were one of the Seven Wonders of the Ancient World listed by Hellenic culture. They were described as a remarkable feat of engineering with an ascending series of tiered gardens containing a wide variety of trees, shrubs, and vines, resembling a large green mountain constructed of mud bricks. It was said to have been built in the ancient city 
Colossus of Rhodes - Score: 0.4841269841269841
Wonders of the Ancient World, it was constructed to celebrate the successful defence of Rhodes city against an attack by Demetrius I of Macedon, who had besieged it for a year with a large army and navy.
Mausoleum at Halicarnassus - Score: 0.4841269841269841
the Greek architects Satyros and Pythius of Priene. Its elevated tomb structure is derived from the tombs of neighbouring Lycia, a territory Mausolus had invaded and annexed c. 360 BC, such as the Nereid Monument.

Reply:
My presumption after reading what RRF should do is that we would get a single rank based on the summed reciprocals of the bm25 and semantic search ranks.

Reply:
Yep I get the same - the duplication is down to the text being split into chunks, and each chunk being embedded separately. Couple of ways to avoid that behaviour:
 - Set the `split_length` to a value large enough that the text doesn't actually get split. Issues with this are that you need to make sure you don't surpass the token limit for whatever embedding model you're using, and my own experience is that when the chunk sizes are very large the semantic search doesn't perform as well for short, keyword-style queries.
 - Add a step to filter the results such that each document is only included once

Reply:
Ah thanks for reminding me, this is not an issue introduced by this PR. Yes de-duping the results would be the way to go I think. 

Adjacently to this, it could be a good sense check to ensure the split length is not larger than the model's context window. Though I think we'll be in the same ball park as trying to determine the emdeddings dimensions - most models don't provide this in their metadata.

Reply:
Yeah so on the app side the search results are joined on to the postgres db such that each document will only appear once. For other projects it might be desirable to return all matches within a document - depends on the use case I guess.

Good point, I'll open an issue. But yes you're right that it's something we'd have to look up for each model, and the other complication is that it's based on the number of tokens rather than the number of words.

Reply:
New issue here: https://github.com/ministryofjustice/rd-search-backend/issues/50